cdhweb/pages/management/commands/cleanup_migrated_content.py

```diff
@@ -0,0 +1,99 @@
+import re
+
+from django.core.management.base import BaseCommand
+from wagtail.core.models import Page
+
+
+class Command(BaseCommand):
+
+    # list of regex patterns to replace with a single space
+    re_to_space = [
+        # non-breaking space
+        re.compile(r'&nbsp;'),
+        # two or more whitespaces
+        re.compile(r'\s\s+'),
+    ]
+
+    # list of regex patterns to replace with first match
+    re_replace_g1 = [
+        # spans with no attributes
+        re.compile(r'<span>(((?!</span>).)*)</span>', flags=re.DOTALL),
+        # remove all but one space before open inline tag
+        re.compile(r'[\n ]+( <(a|em|b|strong|i|span))'),
+        # remove white space after end of open inline tag
+        re.compile(r'(<(a|em|b|strong|i|span)[^>]*>)[\n ]+'),
+        # remove white space before end inline tag
+        re.compile(r'[\n ]+(</(a|em|b|strong|i|span)[^>]*>)'),
+        # remove newline or whitespace between inline tag end and punctuation
+        re.compile(r'(</(a|em|b|strong|i|span)>)[\n ]+(?=[.,;])'),
+        # newline + whitespace after opening <p> or <div>
+        re.compile(r'(<p>|<div>)[\n ]+'),
+    ]
+
+    # list of regex patterns to remove
+    re_remove = [
+        # opening html & body tags
+        re.compile(r'^<html>\s*<body>\s'),
+        # closing html & body tags
+        re.compile(r'\s*</body>\s*</html>$'),
+        # empty div
+        re.compile(r'[\n ]+<div>\s*</div>'),
+    ]
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--noact", action="store_true", default=False,
+            help="Don't save any changes to the database"
+        )
+
+    def handle(self, *args, **options):
+        v_normal = 1
+        verbosity = options['verbosity']
+        noact = options['noact']
+        inline_styles = []
+        for page in Page.objects.all():
+            realpage = page.get_specific()
+            # skip pages with no body content
+            if not hasattr(realpage, 'body'):
+                continue
+            for block in realpage.body:
+                # assuming only one migrated block per page
+                if block.block_type == 'migrated':
```

Comment:
only place this doesn't hold true that I can think of is the form pages (consult/cosponsor); I think I added the iframes to a separate migrated block after the content. but also, is this really assuming that? doesn't look like it is

Reply:
🤔 it's assuming that because I'm saving and publishing the document after cleaning up the one block. I didn't know those had a separate block. Should I set a flag if something needs to be saved and then clean after all looping through all the blocks?

Reply:
imo not worth it; there are only (maybe) two pages that are an exception and they don't have much text